plugins/catalog-backend/src/database/types.ts

```diff
@@ -97,6 +97,20 @@ export type Database = {
    */
   addEntity(tx: unknown, request: DbEntityRequest): Promise<DbEntityResponse>;
 
+  /**
+   * Adds a set of new entities to the catalog.
+   *
+   * @param tx An ongoing transaction
+   * @param request The entities being added
+   */
+  addEntities(
+    tx: unknown,
+    request: {
+      locationId?: string;
```

Comment:
Huh, should we not support batching of entities from different locations too? Not seeing the benefit of that limitation

Reply:
Because of the `entities(<filter>)` that is part of it - with fixed kind/namespace + list of names - it was just a convenient slicing at this point, and makes this parameter list a bit nicer

Reply:
Not following. I'd suggest using `{locationId?: string; entity: Entity}[]` instead

Reply:
Because of this.

https://github.com/spotify/backstage/pull/2810/files#diff-767c1141e0f743ede0e88f680b19204fR264

Somebody will have to do the split.

If you want the split to happen inside the method (there are no callers that desire it at the moment) instead, sure :)

Reply:
Oh haha, sorry I should have looked harder at my own code :) Ok I see what can be done i think

Reply:
fixed!